docs/batch-local-execution-mode.md

```diff
@@ -0,0 +1,90 @@
+# Design: Local execution mode for online serving queries
+
+## Background
+
+The query engine of risingwave is expected to support two kinds of query: highly concurrent point query and adhoc query.
```

Comment:
-> The query engine of RisingWave supports two types of queries: highly concurrent point queries and ad-hoc queries.